dd-java-agent/instrumentation/netty-4.1/src/main/java/datadog/trace/instrumentation/netty41/NettyHttp2Helper.java

```diff
@@ -0,0 +1,39 @@
+package datadog.trace.instrumentation.netty41;
+
+import io.netty.channel.ChannelHandler;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Field;
+
+public class NettyHttp2Helper {
+  private static Class HTTP2_CODEC_CLS;
+  private static MethodHandle IS_SERVER_FIELD;
+
+  static {
+    try {
+      HTTP2_CODEC_CLS =
+          Class.forName(
+              "io.netty.handler.codec.http2.Http2StreamFrameToHttpObjectCodec",
+              false,
+              NettyHttp2Helper.class.getClassLoader());
+      Field f = HTTP2_CODEC_CLS.getDeclaredField("isServer");
+      f.setAccessible(true);
+      IS_SERVER_FIELD = MethodHandles.lookup().unreflectGetter(f);
+    } catch (Throwable t) {
+      HTTP2_CODEC_CLS = null;
+      IS_SERVER_FIELD = null;
+    }
```

Comment:
Introducing some local variables here lets you keep the static variables `final` without the compiler complaining.
```suggestion
    Class codecClass;
    MethodHandle isServerField;
    try {
      codecClass =
          Class.forName(
              "io.netty.handler.codec.http2.Http2StreamFrameToHttpObjectCodec",
              false,
              NettyHttp2Helper.class.getClassLoader());
      Field f = codecClass.getDeclaredField("isServer");
      f.setAccessible(true);
      isServerField = MethodHandles.lookup().unreflectGetter(f);
    } catch (Throwable t) {
      codecClass = null;
      isServerField = null;
    }
    HTTP2_CODEC_CLS = codecClass;
    IS_SERVER_FIELD = isServerField;
```

Reply:
Also do we want to have a `debug` log message in the `catch` block, so we can tell if there was an issue with the reflection calls?

Reply:
Yes I'll add a debug but not logging if CNF because it can be a nominal case (no http2 codec on the classpath)